Fixing mysql ALTER TABLE IF NOT EXISTS when upgrading

sc979 opened the pull request:
# Pull Request Template

## Description

As "ALTER TABLE 'xxxx' IF NOT EXISTS" is not supported by mysql, this PR is intended to check if a column exist before trying to alters it.
![image](https://user-images.githubusercontent.com/34628915/62630866-05e56800-b930-11e9-915f-190330a0e53e.png)



returns :
- 0 : if it needs to be altered,
- 1 : if it's already done (eg: in a previous upgrade script),
- -1 : on errors

**Fixes** # (none)

## Type of change

- [x] Patch fixing an issue (non-breaking change)

## Target serie

- [x] 19.04.x -> to be adapted (remove 19.10 script,...)
- [x] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

- Install a 19.04.0 with a mariaDB database and try to update to the latest version
- Install a 19.04.0 with a deported mysql database and try to update to the latest version
The scripts should not fail at all.

## Checklist

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

lgcosta commented on www/class/centreonDB.class.php:
```diff
@@ -368,7 +370,7 @@ public function getProperties()
             $version = $row['mysql_version'];
             $info['version'] = $row['mysql_version'];
-            if ($DBRESULT = $this->query("SHOW TABLE STATUS FROM `".$this->dsn['database']."`")) {
+            if ($DBRESULT = $this->query("SHOW TABLE STATUS FROM `" . $this->dsn['database'] . "`")) {
```
`s/$DBRESULT/$DbResult/`

lgcosta commented on www/class/centreonDB.class.php:
```diff
@@ -368,7 +370,7 @@ public function getProperties()
             $info['version'] = $row['mysql_version'];
-            if ($DBRESULT = $this->query("SHOW TABLE STATUS FROM `".$this->dsn['database']."`")) {
+            if ($DBRESULT = $this->query("SHOW TABLE STATUS FROM `" . $this->dsn['database'] . "`")) {
                 while ($data = $DBRESULT->fetch()) {
```
`s/$DBRESULT/$DbResult/`

sc979 commented:
@lgcosta , changes done ;)

ganoze commented on www/class/centreonDB.class.php:
```diff
@@ -390,4 +392,50 @@ public function getProperties()
+    public function isColumnExist(string $table = null, string $column = null): int
+    {
+        if (!$table || !$column) {
+            $this->log->insertLog(2, 'UPGRADE PROCESS : Missing data to check if column exist');
```
IMHO as you are within the global CentreonDB class, logs should be handled by the calling code, not directly by the class itself.

sc979 replied:
removed ;)

miza92 commented on www/class/centreonDB.class.php:
```diff
@@ -390,4 +392,48 @@ public function getProperties()
+            $stmt->bindValue(':tableName', $table, \PDO::PARAM_STR);
+            $stmt->bindValue(':columnName', $column, \PDO::PARAM_STR);
+            $stmt->execute();
+            $stmt->fetch();
```
```suggestion
```

miza92 replied:
not needed

sc979 commented:
Thanks @miza92 for your suggestion. I applied it ;)

sc979 commented:
rebased on each specific branches and added to these PR :
https://github.com/centreon/centreon/pull/7801
https://github.com/centreon/centreon/pull/7799
https://github.com/centreon/centreon/pull/7798
